src/user_input_compiler.rs

```diff
@@ -298,7 +291,9 @@ fn match_timesuffix(s :&mut Chars) -> Result<i64, Input> {
   Err( InvalidSyntax(format!("Invalid TimeSuffix")) )
 }
 
-fn starts_with(it :&mut Chars, con :&str) -> bool {
+
+/// Search without modifing the Iterator
+fn starts_with(it :&Chars, con :&str) -> bool {
```

Comment:
What does `con` stand for? Variable name could be made easier to understand. Same for the parameter in `matches_with`.